### DISCUSSION TITLE
Github file size limit issue

### DISCUSSION THREAD

**Discussion by raysefo** [community/community#147125] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi,
I am trying to push changes to my repository. I have already deleted a 7z file which you can see in the outgoing as **zip deleted** message. Since there is no 7z file on my local machine, how can I push the changes?

![Ekran görüntüsü 2024-12-15 090715](https://github.com/user-attachments/assets/54b24755-11a1-491d-ae94-63e576243e15)

**Comment by spenserblack** [comment, marked as answer]
Even though you might have deleted the file, it's still in your *history* that you're trying to push up to GitHub. Git tracks changes, and all those changes added together become the files you see in your project. So it sounds like you have one really large commit that adds that file, and a later commit that deletes it.

There are several ways to fix this (like using an interactive rebase to edit commits), but I'll recommend the most beginner-friendly approach: clone your repo to a new location, re-apply your changes from your other cloned repo (making sure not to commit large and unnecessary files), and push.

**Reply by Logophoman** [reply]
Yepp. Or you delete that file permanently from Git tracking but then you better be sure.
